File: conformance/tests/boolean-logic/boolean-logic.tsop.go
package contract

import "tsop"

type BooleanLogic struct {
	tsop.SmartContract
	Threshold tsop.Int `tsop:"readonly"`
}

func (c *BooleanLogic) Verify(a tsop.Int, b tsop.Int, flag tsop.Bool) {
	aAboveThreshold := a > c.Threshold
	bAboveThreshold := b > c.Threshold
	bothAbove := aAboveThreshold && bAboveThreshold
	eitherAbove := aAboveThreshold || bAboveThreshold
	notFlag := !flag
	tsop.Assert(bothAbove || (eitherAbove && notFlag))
}

File: conformance/tests/boolean-logic/boolean-logic.tsop.rs
use tsop::prelude::*;

#[tsop::contract]
struct BooleanLogic {
    #[readonly]
    threshold: Int,
}

#[tsop::methods(BooleanLogic)]
impl BooleanLogic {
    #[public]
    fn verify(&self, a: Int, b: Int, flag: bool) {
        let a_above_threshold = a > self.threshold;
        let b_above_threshold = b > self.threshold;
        let both_above = a_above_threshold && b_above_threshold;
        let either_above = a_above_threshold || b_above_threshold;
        let not_flag = !flag;
        assert!(both_above || (either_above && not_flag));
    }
}
